crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -548,17 +548,33 @@ async fn get_quote(
         // verified quote here on purpose.
         verification: Default::default(),
     };
-    let fee_amount = match order_data.fee_amount.is_zero() {
-        // If an ETHFlow order was created with 0 fee it means it has the semantics
-        // of a limit order where solvers are responsible of baking in an appropriate execution fee
-        // into their limit prices. In that case we don't require that the fee amount matches one of
-        // our stored quotes.
-        true => None,
-        false => Some(order_data.fee_amount),
-    };
```

Comment:
This logic is covered by https://github.com/cowprotocol/services/blob/ccb44effb1de61e68a6f7ce63875e5f624b95301/crates/shared/src/order_validation.rs#L831-L833